src/status_im2/contexts/shell/animation.cljs

```diff
@@ -1,177 +1,85 @@
+    (utils/change-selected-stack-id stack-id true home-stack-state-value)
+    (js/setTimeout
+     (fn []
+       (utils/load-stack stack-id)
+       (utils/change-shell-status-bar-style))
+     (if animate? shell.constants/shell-animation-time 0))))
 
 (defn change-tab
   [stack-id]
-  (reanimated/set-shared-value (:animate-home-stack-left @shared-values-atom) false)
-  (reanimated/set-shared-value (:selected-stack-id @shared-values-atom) (name stack-id))
-  (change-selected-stack-id stack-id true))
+  (reanimated/set-shared-value (:animate-home-stack-left @state/shared-values-atom) false)
+  (reanimated/set-shared-value (:selected-stack-id @state/shared-values-atom) (name stack-id))
+  (utils/load-stack stack-id)
+  (utils/change-selected-stack-id stack-id true))
 
 (defn bottom-tab-on-press
-  [stack-id]
-  (when (and @shared-values-atom (not= stack-id @selected-stack-id))
-    (let [stack-load-delay (if (home-stack-open?)
-                             0
-                             shell.constants/shell-animation-time)]
-      (if (home-stack-open?)
-        (change-tab stack-id)
-        (open-home-stack stack-id true))
-      (js/setTimeout #(load-stack stack-id) stack-load-delay))))
+  [stack-id animate?]
+  (when (and @state/shared-values-atom (not= stack-id @state/selected-stack-id))
+    (if (utils/home-stack-open?)
+      (change-tab stack-id)
+      (open-home-stack stack-id animate?))
+    (when animate? (utils/update-view-id))))
 
 (defn close-home-stack
   [animate?]
   (let [stack-id               nil
-        home-stack-state-value (calculate-home-stack-state-value stack-id animate?)]
-    (reanimated/set-shared-value (:animate-home-stack-left @shared-values-atom) true)
-    (reanimated/set-shared-value (:home-stack-state @shared-values-atom) home-stack-state-value)
-    (change-selected-stack-id stack-id true home-stack-state-value)
-    (change-shell-status-bar-style)))
+        home-stack-state-value (utils/calculate-home-stack-state-value stack-id animate?)]
+    (reanimated/set-shared-value (:animate-home-stack-left @state/shared-values-atom) true)
+    (reanimated/set-shared-value (:home-stack-state @state/shared-values-atom) home-stack-state-value)
+    (utils/change-selected-stack-id stack-id true home-stack-state-value)
+    (utils/change-shell-status-bar-style)
+    (when animate? (utils/update-view-id))))
+
+;; Floating Screen
+(defn animate-floating-screen
+  [screen-id {:keys [id animation community-id hidden-screen?]}]
+  (let [opening-screen? (not (utils/floating-screen-open? screen-id))]
+    (when (utils/floating-screen-animate? screen-id animation)
+      ;; Animate Floating Screen
+      (reanimated/set-shared-value
+       (get-in @state/shared-values-atom [screen-id :screen-state])
+       animation)
+      (reset! state/floating-screens-state
+        (assoc @state/floating-screens-state screen-id animation)))
+    (js/setTimeout
```

Comment:
Hi, the timer is only 200 ms. So in most cases, it will always run before the screen is unmounted. And even if we somehow close it before 200 ms, we still need to dispatch events, to accordingly update shell card position, etc (as the screen is opened).